pkg/apis/aadpodidentity/types.go

```diff
@@ -5,39 +5,67 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// EventType is a type that represents critical events that are sent to MIC.
 type EventType int
 
 const (
-	PodCreated      EventType = 0
-	PodDeleted      EventType = 1
-	PodUpdated      EventType = 2
+	// PodCreated is an event that is sent to the event channel when a pod is created.
+	PodCreated EventType = 0
+
+	// PodDeleted is an event that is sent to the event channel when a pod is deleted.
+	PodDeleted EventType = 1
+
+	// PodUpdated is an event that is sent to the event channel when a pod is updated.
+	PodUpdated EventType = 2
+
+	// IdentityCreated is an event that is sent to the event channel when an AzureIdentity is created.
 	IdentityCreated EventType = 3
+
+	// IdentityDeleted is an event that is sent to the event channel when an AzureIdentity is deleted.
 	IdentityDeleted EventType = 4
+
+	// IdentityUpdated is an event that is sent to the event channel when an AzureIdentity is updated.
 	IdentityUpdated EventType = 5
-	BindingCreated  EventType = 6
-	BindingDeleted  EventType = 7
-	BindingUpdated  EventType = 8
-	Exit            EventType = 9
+
+	// BindingCreated is an event that is sent to the event channel when an AzureIdentityBinding is created.
+	BindingCreated EventType = 6
+
+	// BindingDeleted is an event that is sent to the event channel when an AzureIdentityBinding is deleted.
+	BindingDeleted EventType = 7
+
+	// BindingUpdated is an event that is sent to the event channel when an AzureIdentityBinding is updated.
+	BindingUpdated EventType = 8
+
+	// Exit is an event that is sent to the event channel when the program exits.
+	Exit EventType = 9
 )
 
 const (
-	CRDGroup    = "aadpodidentity.k8s.io"
-	CRDVersion  = "v1"
+	// CRDGroup is the group name of aad-pod-identity CRDs.
+	CRDGroup = "aadpodidentity.k8s.io"
+
+	// CRDVersion is the version of the CRD group.
+	CRDVersion = "v1"
+
+	// CRDLabelKey is the
 	CRDLabelKey = "aadpodidbinding"
 
+	// BehaviorKey is the key that describes the behavior of aad-pod-identity.
```

Comment:
Can we add - Supported values: `namespaced`, used for running in namespaced mode. AzureIdentity, AzureIdentityBinding and pod in the same namespace will only be matched for this behavior